butterfree/migrations/migration.py

```diff
@@ -0,0 +1,36 @@
+"""Migration entity."""
+
+from abc import ABC, abstractmethod
+from typing import Any, Callable
+
+
+class Migration(ABC):
+    """Abstract base class for Migrations."""
+
+    @abstractmethod
+    def create_query(self) -> Any:
```

Comment:
`create_query` given what?

Reply:
I've updated this as well. Just to let you know, I've took some time to get back to this PR because I was implementing these methods and I've found a similar structure to the one you suggested, that's nice!